os/arch/arm/src/imxrt/imxrt_usbhost.c

```diff
@@ -0,0 +1,265 @@
+
+	return 0;
+}
+
+/************************************************************************************
+ * Public Functions
+ ************************************************************************************/
+
+/************************************************************************************
+ * Name: imxrt_usbhost_bootinitialize
+ *
+ * Description:
+ *   Called from imxrt_boardinitialize very early in inialization to setup USB
+ *   host-related GPIO pins for the LPC-H3131 board.
+ *
+ *   SIGNAL      GPIO
+ *   ----------- -------
+ *   #OTG_PWR_E  GPIO19
+ *   #OTG_OVRCR  GPIO20
+ *
+ ************************************************************************************/
+
+void weak_function imxrt_usbhost_bootinitialize(void)
+{
+	/* Configure output pin to drive VBUS power (initial state: power off) */
+
+	imxrt_config_gpio(GPIO_USB_OTG1_PWR_1);
+
+	/* Configure input pin to detect overrcurrent errors */
+
+	imxrt_config_gpio(GPIO_USB_OTG1_OC_1);
+
+	/* Configure to receive interrupts on the overrcurrent input pin */
+#warning Missing logic
+}
+
+/***********************************************************************************
+ * Name: imxrt_usbhost_initialize
+ *
+ * Description:
+ *   Called at application startup time to initialize the USB host functionality.
+ *   This function will start a thread that will monitor for device
+ *   connection/disconnection events.
+ *
+ ***********************************************************************************/
+
+int imxrt_usbhost_initialize(void)
+{
+	pid_t pid;
+	int ret;
+
+	/* Then get an instance of the USB EHCI interface. */
+
+	g_ehciconn = imxrt_ehci_initialize(0);
+	if (!g_ehciconn) {
+		IMXLOG("ERROR: imxrt_ehci_initialize failed\n");
+		return -ENODEV;
+	}
+
+	/* Start a thread to handle device connection. */
+	pid = task_create("EHCI Monitor", CONFIG_USBHOST_DEFPRIO, CONFIG_USBHOST_STACKSIZE, (main_t) ehci_waiter, (FAR char *const *)NULL);
+	if (pid < 0) {
+		IMXLOG("ERROR: Failed to create ehci_waiter task: %d\n");
+		return -ENODEV;
+	}
+
+	return OK;
+}
+
+/***********************************************************************************
+ * Name: imxrt_usbhost_vbusdrive
+ *
+ * Description:
+ *   Enable/disable driving of VBUS 5V output.  This function must be provided by
+ *   each platform that implements the OHCI or EHCI host interface
+ *
+ * Input Parameters:
+ *   rhport - Selects root hub port to be powered host interface. Since the LPC31
+ *     has only a downstream port, zero is the only possible value for this
+ *     parameter.
+ *   enable - true: enable VBUS power; false: disable VBUS power
+ *
+ * Returned Value:
+ *   None
+ *
+ ***********************************************************************************/
+
+void imxrt_usbhost_vbusdrive(int rhport, bool enable)
+{
+	uinfo("RHPort%d: enable=%d\n", rhport + 1, enable);
+
+	/* The LPC3131 has only a single root hub port */
+
+#if 0
+	if (rhport == 0) {
```

Comment:
Please remove the code if not required.

Reply:
I want to keep this code for reference.
I will add another definition for it
Please check it